sound/soc/sof/loader.c

```diff
@@ -106,6 +106,12 @@ int snd_sof_parse_module_memcpy(struct snd_sof_dev *sdev,
 	/* module->size doesn't include header size */
 	remaining = module->size;
 	for (count = 0; count < module->num_blocks; count++) {
+		/* check for wrap */
+		if (remaining < sizeof(*block)) {
+			dev_err(sdev->dev, "error: not enough data remaining\n");
+			return -EINVAL;
+		}
+
```

Comment:
i'd move the test line 117 to line 156. The code as is looks like a mistaken copy-paste, when it's really doing the check earlier than needed (and other errors could be trapped first before decrementing line 156)

Reply:
ok